solutions/cf-1566B.mdx

```diff
@@ -0,0 +1,68 @@
+---
+id: cf-1566B
+source: CF
+title: MIN-MEX Cut
+author: Ryan Chou
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/94803)
+
+
+# Explanation
+
+Since the MEX can be at most 2 when there is a substring that contains a 0 and 1, there's only 3 possible cases. If there are no zeros, then the MEX is 0, since all substrings would be missing a 0.  If all zeros are adjacent to one another, we can cut the entire sequence out, leaving a MEX of 1. The other groups, all comprised of ones would have a MEX of 0. Leaving the answer to be $max(0, 1) = 1$. Lastly, if there are zeros and they're not adjacent to one another, then a substring like this must exist, {0, 1}, resulting in a MEX of 2.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <algorithm>
+using namespace std;
+
```

Comment:
is there a reason why this doesn't compile when I try to submit? 

https://codeforces.com/contest/1566/submission/129216986

Reply:
i think it's something that sans suggested, let me fix it